### PR TITLE
Add differentiability indicators to reduction operators, needed for ONNX Training

### PR DESCRIPTION
**Pull request by sveta-levitan**
Still need to update documentation.

Signed-off-by: Svetlana Levitan <sveta_levitan@yahoo.com>

### CODE REVIEW

**Comment by sveta-levitan** [comment]
@wschin Please add documentation and label this for 1.8 release. Thank you!

**Comment by wschin** [comment]
> @wschin Please add documentation and label this for 1.8 release. Thank you!

I generated the new docs using this commit but I found one place we need to fix. `ArgMax` and `ArgMin` should be non-differentiable at input and output.

File: onnx/defs/reduction/defs.cc [hunk 1]
```diff
@@ -220,12 +239,24 @@ The type of the output tensor is integer.)DOC";
...
-    schema.Input(0, "data", "An input tensor.", "T");
+    schema.Input(
+        0, 
+        "data", 
+        "An input tensor.", 
+        "T",
+        OpSchema::Single,
+        true,
+        1,
+        OpSchema::Differentiable);
```
Comment (wschin) [inline onnx/defs/reduction/defs.cc:250]:
```suggestion
        OpSchema::NonDifferentiable);
```

File: onnx/defs/reduction/defs.cc [hunk 2]
```diff
@@ -220,12 +239,24 @@ The type of the output tensor is integer.)DOC";
...
     schema.Output(
         0,
         "reduced",
         "Reduced output tensor with integer data type.",
-        "tensor(int64)");
+        "tensor(int64)",
+        OpSchema::Single,
+        true,
+        1,
+        OpSchema::Differentiable);
```
Comment (wschin) [inline onnx/defs/reduction/defs.cc:259]:
```suggestion
        OpSchema::NonDifferentiable);
```

**Review by wschin — APPROVED** [review]

**Pull request merged**